src/DotNetNuke.PowerBI/Services/EmbedService.cs

```diff
@@ -462,17 +469,44 @@ private async Task<EmbedToken> GenerateTokenAsync(string username, string roles,
                             rls.Roles = rolesList;
                         }
                         // Generate Embed Token with effective identities.
-                        generateTokenRequestParameters = new GenerateTokenRequest(accessLevel: "view", identities: new List<EffectiveIdentity> { rls });
+
+                        if (hasEditPermission)
```

Comment:
Remove ifs and use a variable. i.e.

string permission = hastEditPermission ? "edit" : "view";